### PR TITLE
[PM-6383] Migrate clearClipboard account setting to autofill settings service

### PR DESCRIPTION
**Pull request by jprusik**
## Type of change

- Tech debt (refactoring, code cleanup, dependency upgrades, etc)

## Objective

As a follow-on from https://github.com/bitwarden/clients/pull/7767 the account setting `clearClipboard` should be included in the autofill settings service state provider

## Code changes

- Replaced instances of get/set `clearClipboard` state now handled by the Autofill Settings service with their appropriate counterparts
- Created migration script to upgrade or rollback state storage changes

### CODE REVIEW

**Review by cagonzalezcs — DISMISSED** [review]
Looks good from my side, CodeCov is complaining about some missed lines in terms of coverage, but I'm not keen on blocking this because of that.

If you do need a re-approval to address coverage on those changes, let me know.

**Review by cagonzalezcs — APPROVED** [review]

File: libs/common/src/platform/services/state.service.ts
Comment (justindbaur) [inline libs/common/src/platform/services/state.service.ts:1]: Should be able to delete [this](https://github.com/bitwarden/clients/blob/dee0b20554694244bceff489dd7c27228f6b32b9/libs/common/src/platform/models/domain/account.ts#L204) now.

File: libs/common/src/state-migrations/migrations/24-move-clear-clipboard-to-autofill-settings-state-provider.spec.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
+import { any, MockProxy } from "jest-mock-extended";
+
+import { ClearClipboardDelay } from "../../../../../apps/browser/src/autofill/constants";
+import { AutofillOverlayVisibility } from "../../../../../apps/browser/src/autofill/utils/autofill-overlay.enum";
```
Comment (justindbaur) [inline libs/common/src/state-migrations/migrations/24-move-clear-clipboard-to-autofill-settings-state-provider.spec.ts:4]: You shouldn't import `apps/browser` into `libs/common`, even just for a test. I would just represent these values like the jsonified version of themselves like how they are saved in state.

**Review by justindbaur — CHANGES_REQUESTED** [review]
Couple minor things.

**Review by cagonzalezcs — DISMISSED** [review]

**Review by justindbaur — APPROVED** [review]

**Pull request merged**